packages/next/client/components/app-router.tsx

```diff
@@ -195,10 +197,11 @@ function Router({
 
     return {
       // This is turned into a readonly class in `useSearchParams`
-      searchParams: url.searchParams,
+      // on `dynamic = 'force-static`, return empty search params
+      searchParams: forceStatic ? new URLSearchParams() : url.searchParams,
```

Comment:
This seems to be handled incorrectly as you can navigate from a page with `forceStatic` `true` to a page that is dynamic, or has `forceStatic` `false`. In that case searchParams will never be provided even though it should be provided. This would also mean that `forceStatic` has to be returned as part of the RSC payload when client-side navigating in order to make it consistent, however I think in those case we want searchParams to be available client-side.

Reply:
@timneutkens added test for the navigation case, obv failing right now. You think I should skip `force-static` tests and only get the bailout correct in this PR? `force-static` isn't working currently anyway for `useSearchParams` so could be fixed later.

@huozhi you think your work with `next/dynamic` could be reused here somehow? Like reusing `NoSSR` instead of creating a new one?

Reply:
there's a `suspense()` api exported in dynamic-no-ssr.tsx, do you feel you can leverage that? it's also already been handled on client.

```js
if (staticGenerationStore?.isStaticGeneration) suspense()
```

wdyt? does the API name make sense for this case? open to rename it if necessary